src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,112 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [validImg, setValidImg] = useState(true);
+  const [validImdb, setValidImdb] = useState(true);
+
+  function handleTextField(setText: (value: string) => void) {
+    const handleText = (value: string) => {
+      setText(value);
+    };
+
+    return handleText;
+  }
+
+  let disabledButton = true;
+
+  if (title && imgUrl && imdbUrl && imdbId && validImg === true
+    && validImdb === true) {
+    disabledButton = false;
```

Comment:
```suggestion
  const canSubmit = title && imgUrl && imdbUrl && imdbId && validImg
    && validImdb;
```
Just save the condition as variable. Also, there is no need to compare the boolean value with true or false. Use boolean/!boolean instead

Reply:
There is a TS Error if I change condition to variable.